app/scripts/controllers/buildConfig.js

```diff
@@ -74,6 +109,7 @@ angular.module('openshiftConsole')
                 };
               }
               $scope.buildConfig = buildConfig;
+              $scope.updatedBuildConfig = angular.copy(buildConfig);
```

Comment:
That's what we do elsewhere. It would be nice to give the user a warning though before they go through the trouble editing.